File: P02/PE_Problem02.js
/*
	Project Euler
	Vadim Brodsky
	Friday - August 31 2012
	Problem 02
*/
"use strict";
var timer = require("../timer.js");
timer.startTimer();

var evenFib = function (limit){
	var num1 = 0;
	var num2 = 1;
	var sum = 0;
	var evenSum = 0;
	while ((num1 + num2) < limit) {
		sum = num1 + num2;
		if (isEven(sum)) {
			evenSum += sum;
		}
		console.log(sum);
		num1 = num2;
		num2 = sum;
	}
	return evenSum;
};

var isEven = function (number) {
	if (number % 2 === 0) {
		return true;
	}	
};

console.log("Sum of the Even Fibonacci numbers: " + evenFib(4000000));	
timer.endTimer();

File: P02/PE_Problem02.rb
require 'benchmark'

def even_fibonacci_numbers
  limit = 4000000
  sum = 0

  current_number = 0
  num1 = 1
  num2 = 1

  while current_number < limit
    current_number = num1 + num2
    num1 = num2
    num2 = current_number

    # puts current_number

    if current_number.even?
      sum = sum + current_number
    end 
  end

  return sum
end


# Code benchmark
Benchmark.bm(5) do |bm|
  bm.report('V1: ') do
    even_fibonacci_numbers
  end
end

puts "Answer: #{even_fibonacci_numbers}"